### ISSUE TITLE
labeled training data

### ISSUE THREAD

**Issue by bhomass** [yueyu1030/COSINE#27]
isn't whole idea of COSINE, so you would not require manual annotation, and the so called init step uses weak learning (says so explicitly in the paper). So why do you need the ground truth label in the training data?

**Comment by bhomass** [comment]
from what I can see, not 100% sure, but only the label from dev and test data are used, and only used in eval to see how much the model has converged during self-training, while the label from train data is not used. Is this true? 

So therefore, we are expected to provide at least some labeled samples for this purpose?

**Comment by bhomass** [comment]
could I also ask why two separate data files, one for dev and the other for test, since you seemed alway evaluate both one after the other.

**Comment by yueyu1030** [comment]
> could I also ask why two separate data files, one for dev and the other for test, since you seemed alway evaluate both one after the other. and is it necessary to create over 1000 samples for each category? That's still very substantial amount of manual labor.



> from what I can see, not 100% sure, but only the label from dev and test data are used, and only used in eval to see how much the model has converged during self-training, while the label from train data is not used. Is this true?
> 
> So therefore, we are expected to provide at least some labeled samples for this purpose?
Yes, in this work, we still use some labeled data for *model checkpoint selection* only. Overall I think we may not need massive number of labeled data for the development set. In some cases 100 or less is enough.
